### PR TITLE
Add miniapp for tridiagonal eigensolver (and plotting facilities)

### LINKED ISSUE #525: Miniapp tridiagonal eigensolver
Depends on #490.

### PR DESCRIPTION
**Pull request by albestro**
Close #525 

- [x] Test plot script
- [x] ~Add check in miniapp~
- [x] Feedback on naming (trisolver is ok? stev? any other suggestion?)

### CODE REVIEW

**Comment by rasolca** [comment]
The name of the miniapp should reflect the name of the function (i.e. `tridiagSolver` => `miniapp_tridiag_solver`).
In the plotting script I would use `trid_evp` (as suggested by the TODO comments)

File: miniapp/miniapp_tridiag_solver.cpp [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+    auto tridagReference = [&diag_arr, &offdiag_arr](GlobalElementIndex i) {
+      const size_t row_index = dlaf::to_sizet(i.row());
+      if (i.col() == 0) {
+        return diag_arr[row_index];
+      }
+      else {
+        return offdiag_arr[row_index];
+      }
+    };
```
Comment (rasolca) [inline miniapp/miniapp_tridiag_solver.cpp:115]: Isn't it better to create tridiag_reference as a (m x 2) random matrix?

File: miniapp/miniapp_tridiag_solver.cpp [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+using dlaf::SizeType;
+using dlaf::TileElementSize;
+using dlaf::comm::Communicator;
+using dlaf::comm::CommunicatorGrid;
+using dlaf::common::Ordering;
+using dlaf::matrix::Distribution;
+using dlaf::matrix::MatrixMirror;
+
+struct Options
+    : dlaf::miniapp::MiniappOptions<dlaf::miniapp::SupportReal::Yes, dlaf::miniapp::SupportComplex::Yes> {
```
Comment (rasolca) [inline miniapp/miniapp_tridiag_solver.cpp:51]: Support complex?

File: miniapp/miniapp_tridiag_solver.cpp (hunk 1 above)
Comment (albestro) [reply miniapp/miniapp_tridiag_solver.cpp]: see https://github.com/eth-cscs/DLA-Future/pull/753/commits/7761e7988caeaa02330bf8c7f7ebc49645272a08

File: miniapp/miniapp_tridiag_solver.cpp (hunk 2 above)
Comment (albestro) [reply miniapp/miniapp_tridiag_solver.cpp]: As discussed offline, I disabled complex type.

See https://github.com/eth-cscs/DLA-Future/pull/753/commits/0654e366a95f7c7ae1d9c05066b7156f5504019b

**Review by rasolca — APPROVED** [review]

**Comment by rasolca** [comment]
cscs-ci run

**Pull request merged**